src/profile-logic/profile-data.js

```diff
@@ -361,6 +481,81 @@ export function getSamplesSelectedStates(
   return result;
 }
 
+/**
+ * Go through the samples, and determine their current state.
+ *
+ * For samples that are neither 'FILTERED_OUT_*' nor 'SELECTED', this function compares
+ * the sample's call node to the selected call node, in tree order. It uses the same
+ * ordering as the function compareCallNodes in getTreeOrderComparator.
```

Comment:
This comment needs to be adjusted. For unselected samples you're just using 'AFTER_SELECTED'.